### ISSUE TITLE
`tpolecatOptionsMode` is ignored in test scope

### ISSUE THREAD

**Issue by mbesida** [typelevel/sbt-tpolecat#78]
In my `build.sbt` file I have tpolecat settings defined as following
```
lazy val tpolecatSettings = Seq(
  Compile / tpolecatOptionsMode := ReleaseMode,
  Compile / tpolecatScalacOptions ++= Set(
    new ScalacOption(List("-Ypatmat-exhaust-depth", "off")),
    ScalacOptions.warnOption("""conf:cat=deprecation&site=com\.foo\.bar\..*:s""")
  ),
  Compile / doc / tpolecatOptionsMode := DevMode,
  Test / tpolecatOptionsMode := DevMode
)
```
which are then added to a project.
In version `0.2.2` it worked as expected: `fatal-warnings` was present in compile scope but absent in test scope(as it is defined in `tpolecatScalacOptions` for `DevMode` and `ReleaseMode`).
Then I changed the plugin version to `0.2.3` and test scope started to have `fatal-warnings`, meaning that `sbt test` fails due to compilation errors, because there are warnings in test scope. I also tried the most recent version of the plugin(0.3.1 at the moment of creating this issue), the result is the same.

Output of `inspect Test/tpolecatScalacOptions` when using plugin version `0.2.3`
```
sbt:server> inspect Test/tpolecatScalacOptions
[info] Setting: scala.collection.immutable.Set[io.github.davidgregory084.ScalacOption] = ListSet(ScalacOption(-encoding utf8), ScalacOption(-deprecation), ScalacOption(-feature), ScalacOption(-unchecked), ScalacOption(-language:existentials), ScalacOption(-language:experimental.macros), ScalacOption(-language:higherKinds), ScalacOption(-language:implicitConversions), ScalacOption(-Xcheckinit), ScalacOption(-Xlint), ScalacOption(-Xlint:adapted-args), ScalacOption(-Xlint:by-name-right-associative), ScalacOption(-Xlint:constant), ScalacOption(-Xlint:delayedinit-select), ScalacOption(-Xlint:deprecation), ScalacOption(-Xlint:doc-detached), ScalacOption(-Xlint:implicit-recursion), ScalacOption(-Xlint:implicit-not-found), ScalacOption(-Xlint:inaccessible), ScalacOption(-Xlint:infer-any), ScalacOption(-Xlint:missing-interpolator), ScalacOption(-Xlint:nullary-override), ScalacOption(-Xlint:nullary-unit), ScalacOption(-Xlint:option-implicit), ScalacOption(-Xlint:package-object-classes), ScalacOption(-Xlint:poly-implicit-overload), ScalacOption(-Xlint:private-shadow), ScalacOption(-Xlint:stars-align), ScalacOption(-Xlint:strict-unsealed-patmat), ScalacOption(-Xlint:type-parameter-shadow), ScalacOption(-Xlint:unsound-match), ScalacOption(-Xlint:-byname-implicit), ScalacOption(-Yno-adapted-args), ScalacOption(-Ykind-projector), ScalacOption(-Ywarn-dead-code), ScalacOption(-Ywarn-extra-implicit), ScalacOption(-Ywarn-inaccessible), ScalacOption(-Ywarn-nullary-override), ScalacOption(-Ywarn-nullary-unit), ScalacOption(-Ywarn-numeric-widen), ScalacOption(-Ywarn-unused), ScalacOption(-Ywarn-unused-import), ScalacOption(-Ywarn-value-discard), ScalacOption(-Ywarn-unused:implicits), ScalacOption(-Ywarn-unused:imports), ScalacOption(-Ywarn-unused:locals), ScalacOption(-Ywarn-unused:params), ScalacOption(-Ywarn-unused:patvars), ScalacOption(-Ywarn-unused:privates), ScalacOption(-Wdead-code), ScalacOption(-Wextra-implicit), ScalacOption(-Wnumeric-widen), ScalacOption(-Wvalue-discard), ScalacOption(-Wunused:nowarn), ScalacOption(-Wunused:implicits), ScalacOption(-Wunused:explicits), ScalacOption(-Wunused:imports), ScalacOption(-Wunused:locals), ScalacOption(-Wunused:params), ScalacOption(-Wunused:patvars), ScalacOption(-Wunused:privates))
[info] Description:
[info]  The set of scalac options that will be applied by the sbt-tpolecat plugin.
[info] Provided by:
[info]  ProjectRef(uri("file:/home/mbesida/myproject/"), "root") / Test / tpolecatScalacOptions
[info] Defined at:
[info]  (io.github.davidgregory084.TpolecatPlugin.projectSettings) TpolecatPlugin.scala:141
[info] Dependencies (D=derived from):
[info]  D Test / tpolecatReleaseModeOptions
[info]  D Test / tpolecatCiModeOptions
[info]  D Test / tpolecatDevModeOptions
[info]  D Test / tpolecatOptionsMode
[info] Reverse dependencies (D=derives):
[info]  D Test / scalacOptions
[info] Delegates:
[info]  Test / tpolecatScalacOptions
[info]  Runtime / tpolecatScalacOptions
[info]  Compile / tpolecatScalacOptions
[info]  tpolecatScalacOptions
[info]  ThisBuild / Test / tpolecatScalacOptions
[info]  ThisBuild / Runtime / tpolecatScalacOptions
[info]  ThisBuild / Compile / tpolecatScalacOptions
[info]  ThisBuild / tpolecatScalacOptions
[info]  Zero / Test / tpolecatScalacOptions
[info]  Zero / Runtime / tpolecatScalacOptions
[info]  Zero / Compile / tpolecatScalacOptions
[info]  Global / tpolecatScalacOptions
[info] Related:
[info]  Compile / doc / tpolecatScalacOptions
[info]  tasks / Test / tpolecatScalacOptions
[info]  common / Test / console / tpolecatScalacOptions
[info]  common / Compile / console / tpolecatScalacOptions
[info]  tasks / Compile / console / tpolecatScalacOptions
[info]  tasks / Compile / doc / tpolecatScalacOptions
[info]  serverApi / Test / console / tpolecatScalacOptions
[info]  serverApi / Compile / console / tpolecatScalacOptions
[info]  common / Compile / tpolecatScalacOptions
[info]  Compile / console / tpolecatScalacOptions
[info] ...
```

Output of `inspect tree Test/tpolecatScalacOptions` when using plugin version `0.2.3`
```
sbt:server> inspect tree Test/tpolecatScalacOptions
[info] Test / tpolecatScalacOptions = ListSet(ScalacOption(-encoding utf8), ScalacOpt..
[info]   +-tpolecatCiModeOptions = ListSet(ScalacOption(-encoding utf8), ScalacOption..
[info]   +-tpolecatDevModeOptions = ListSet(ScalacOption(-encoding utf8), ScalacOptio..
[info]   +-Test / tpolecatOptionsMode = DevMode
[info]   +-tpolecatReleaseModeOptions = ListSet(ScalacOption(-encoding utf8), ScalacO..
[info]   
```

Output of `inspect Test/tpolecatScalacOptions` when using plugin version `0.2.2`
```
sbt:server> inspect Test/tpolecatScalacOptions
[info] Setting: scala.collection.immutable.Set[io.github.davidgregory084.ScalacOption] = ListSet(ScalacOption(-encoding utf8), ScalacOption(-deprecation), ScalacOption(-feature), ScalacOption(-unchecked), ScalacOption(-language:existentials), ScalacOption(-language:experimental.macros), ScalacOption(-language:higherKinds), ScalacOption(-language:implicitConversions), ScalacOption(-Xcheckinit), ScalacOption(-Xlint), ScalacOption(-Xlint:adapted-args), ScalacOption(-Xlint:by-name-right-associative), ScalacOption(-Xlint:constant), ScalacOption(-Xlint:delayedinit-select), ScalacOption(-Xlint:deprecation), ScalacOption(-Xlint:doc-detached), ScalacOption(-Xlint:implicit-recursion), ScalacOption(-Xlint:implicit-not-found), ScalacOption(-Xlint:inaccessible), ScalacOption(-Xlint:infer-any), ScalacOption(-Xlint:missing-interpolator), ScalacOption(-Xlint:nullary-override), ScalacOption(-Xlint:nullary-unit), ScalacOption(-Xlint:option-implicit), ScalacOption(-Xlint:package-object-classes), ScalacOption(-Xlint:poly-implicit-overload), ScalacOption(-Xlint:private-shadow), ScalacOption(-Xlint:stars-align), ScalacOption(-Xlint:strict-unsealed-patmat), ScalacOption(-Xlint:type-parameter-shadow), ScalacOption(-Xlint:unsound-match), ScalacOption(-Xlint:-byname-implicit), ScalacOption(-Yno-adapted-args), ScalacOption(-Ykind-projector), ScalacOption(-Ywarn-dead-code), ScalacOption(-Ywarn-extra-implicit), ScalacOption(-Ywarn-inaccessible), ScalacOption(-Ywarn-nullary-override), ScalacOption(-Ywarn-nullary-unit), ScalacOption(-Ywarn-numeric-widen), ScalacOption(-Ywarn-unused), ScalacOption(-Ywarn-unused-import), ScalacOption(-Ywarn-value-discard), ScalacOption(-Ywarn-unused:implicits), ScalacOption(-Ywarn-unused:imports), ScalacOption(-Ywarn-unused:locals), ScalacOption(-Ywarn-unused:params), ScalacOption(-Ywarn-unused:patvars), ScalacOption(-Ywarn-unused:privates), ScalacOption(-Wdead-code), ScalacOption(-Wextra-implicit), ScalacOption(-Wnumeric-widen), ScalacOption(-Wvalue-discard), ScalacOption(-Wunused:nowarn), ScalacOption(-Wunused:implicits), ScalacOption(-Wunused:explicits), ScalacOption(-Wunused:imports), ScalacOption(-Wunused:locals), ScalacOption(-Wunused:params), ScalacOption(-Wunused:patvars), ScalacOption(-Wunused:privates))
[info] Description:
[info]  The set of scalac options that will be applied by the sbt-tpolecat plugin.
[info] Provided by:
[info]  ProjectRef(uri("file:/home/mbesida/myproject/"), "root") / Test / tpolecatScalacOptions
[info] Defined at:
[info]  (io.github.davidgregory084.TpolecatPlugin.projectSettings) TpolecatPlugin.scala:141
[info] Dependencies (D=derived from):
[info]  D Test / tpolecatReleaseModeOptions
[info]  D Test / tpolecatCiModeOptions
[info]  D Test / tpolecatDevModeOptions
[info]  D Test / tpolecatOptionsMode
[info] Reverse dependencies (D=derives):
[info]  D Test / scalacOptions
[info] Delegates:
[info]  Test / tpolecatScalacOptions
[info]  Runtime / tpolecatScalacOptions
[info]  Compile / tpolecatScalacOptions
[info]  tpolecatScalacOptions
[info]  ThisBuild / Test / tpolecatScalacOptions
[info]  ThisBuild / Runtime / tpolecatScalacOptions
[info]  ThisBuild / Compile / tpolecatScalacOptions
[info]  ThisBuild / tpolecatScalacOptions
[info]  Zero / Test / tpolecatScalacOptions
[info]  Zero / Runtime / tpolecatScalacOptions
[info]  Zero / Compile / tpolecatScalacOptions
[info]  Global / tpolecatScalacOptions
[info] Related:
[info]  Compile / doc / tpolecatScalacOptions
[info]  tasks / Test / tpolecatScalacOptions
[info]  common / Test / console / tpolecatScalacOptions
[info]  common / Compile / console / tpolecatScalacOptions
[info]  tasks / Compile / console / tpolecatScalacOptions
[info]  tasks / Compile / doc / tpolecatScalacOptions
[info]  serverApi / Test / console / tpolecatScalacOptions
[info]  serverApi / Compile / console / tpolecatScalacOptions
[info]  common / Compile / tpolecatScalacOptions
[info]  Compile / console / tpolecatScalacOptions
[info] ...
```
PS. I have a multimodule project, but `tpolecatSettings` are applied to all of them in the same way

**Comment by DavidGregory084** [comment]
@mbesida I don't seem to be able to replicate this - see #80. Is it possible that there is something else appending to `Test / scalacOptions` in your build? What does the output of `inspect tree Test/scalacOptions` look like?

**Comment by mbesida** [comment]
@DavidGregory084  version of plugin 0.2.3
```
sbt:server> inspect tree Test/scalacOptions
[info] Test / scalacOptions = Task[scala.collection.Seq[java.lang.String]]
[info]   +-Global / autoCompilerPlugins = true
[info]   +-Global / buildDependencies = sbt.internal.BuildDependencies@3c2050c1
[info]   +-Test / configuration = test
[info]   +-Test / earlyOutput = ${BASE}/target/test-early/server-SNAPSHOT.jar
[info]   | +-Test / earlyOutput / artifactPath = target/test-early/server-SNAPSHOT.jar
[info]   | | +-artifact = Artifact(server, jar, jar, None, Vector(), None, Map(), Non..
[info]   | | +-Global / artifactName = sbt.Defaults$$$Lambda$849/1136131052@329ed1f
[info]   | | +-ThisBuild / scalaBinaryVersion = 2.13
[info]   | | +-ThisBuild / scalaVersion = 2.13.8
[info]   | | +-Test / configuration = test
[info]   | | +-crossTarget = target
[info]   | | +-projectID = myproject:SNAPSHOT
[info]   | | 
[info]   | +-ThisBuild / fileConverter = sbt.internal.inc.MappedFileConverter@1d3888b0
[info]   | 
[info]   +-ThisBuild / exportPipelining = false
[info]   +-ThisBuild / fileConverter = sbt.internal.inc.MappedFileConverter@1d3888b0
[info]   +-Global / sbtPlugin = false
[info]   +-ThisBuild / scalaVersion = 2.13.8
[info]   +-Test / scalacOptions / streams = Task[sbt.std.TaskStreams[sbt.internal.uti..
[info]   | +-Global / streamsManager = Task[sbt.std.Streams[sbt.internal.util.Init$Sc..
[info]   | 
[info]   +-Global / semanticdbEnabled = false
[info]   +-Test / semanticdbOptions = List(-Yrangepos, -P:semanticdb:targetroot:/home..
[info]   | +-Test / configuration = test
[info]   | +-ThisBuild / scalaVersion = 2.13.8
[info]   | +-Test / semanticdbTargetRoot = target/test-meta
[info]   |   +-Test / classDirectory = target/test-classes
[info]   |   | +-Test / configuration = test
[info]   |   | +-crossTarget = target
[info]   |   | 
[info]   |   +-Test / configuration = test
[info]   |   +-crossTarget = target
[info]   |   +-Global / semanticdbIncludeInJar = false
[info]   |   
[info]   +-Global / settingsData = Task[sbt.internal.util.Settings[sbt.Scope]]
[info]   +-thisProjectRef = ProjectRef(file:/home/mbesida/myproject/,root)
[info]   +-Test / tpolecatScalacOptions = ListSet(ScalacOption(-encoding utf8), Scala..
[info]   | +-tpolecatCiModeOptions = ListSet(ScalacOption(-encoding utf8), ScalacOpti..
[info]   | +-tpolecatDevModeOptions = ListSet(ScalacOption(-encoding utf8), ScalacOpt..
[info]   | +-Test / tpolecatOptionsMode = DevMode
[info]   | +-tpolecatReleaseModeOptions = ListSet(ScalacOption(-encoding utf8), Scala..
[info]   | 
[info]   +-update = Task[sbt.librarymanagement.UpdateReport]
[info]   
```

```
sbt:server> inspect Test/scalacOptions
[info] Task: scala.collection.Seq[java.lang.String]
[info] Description:
[info]  Options for the Scala compiler.
[info] Provided by:
[info]  ProjectRef(uri("file:/home/mbesida/myproject/"), "root") / Test / scalacOptions
[info] Defined at:
[info]  (sbt.Defaults.configTasks) Defaults.scala:966
[info]  (sbt.Defaults.configTasks) Defaults.scala:973
[info]  (sbt.Classpaths.compilerPluginConfig) Defaults.scala:4094
[info]  (io.github.davidgregory084.TpolecatPlugin.projectSettings) TpolecatPlugin.scala:132
[info]  (sbt.plugins.SemanticdbPlugin.configurationSettings) SemanticdbPlugin.scala:71
[info]  (sbt.plugins.SemanticdbPlugin.configurationSettings) SemanticdbPlugin.scala:79
[info] Dependencies (D=derived from):
[info]    Test / semanticdbEnabled
[info]    Test / thisProjectRef
[info]    Test / settingsData
[info]    Test / sbtPlugin
[info]    Test / update
[info]    Test / configuration
[info]    Test / earlyOutput
[info]    Test / fileConverter
[info]  D Test / tpolecatScalacOptions
[info]    Test / scalacOptions / streams
[info]    Test / semanticdbOptions
[info]    Test / exportPipelining
[info]    Test / autoCompilerPlugins
[info]  D Test / scalaVersion
[info]    Test / buildDependencies
[info] Reverse dependencies:
[info]  Test / bspBuildTargetScalacOptionsItem
[info] Delegates:
[info]  Test / scalacOptions
[info]  Runtime / scalacOptions
[info]  Compile / scalacOptions
[info]  scalacOptions
[info]  ThisBuild / Test / scalacOptions
[info]  ThisBuild / Runtime / scalacOptions
[info]  ThisBuild / Compile / scalacOptions
[info]  ThisBuild / scalacOptions
[info]  Zero / Test / scalacOptions
[info]  Zero / Runtime / scalacOptions
[info]  Zero / Compile / scalacOptions
[info]  Global / scalacOptions
[info] Related:
[info]  common / Compile / scalacOptions
[info]  common / Test / doc / scalacOptions
[info]  serverApi / Test / scalacOptions
[info]  Test / doc / scalacOptions
[info]  Compile / scalacOptions
[info]  serverApi / Compile / console / scalacOptions
[info]  tasks / Compile / scalacOptions
[info]  serverApi / Compile / scalacOptions
[info]  Compile / console / scalacOptions
[info]  common / Test / scalacOptions
[info] ...
```

**Comment by DavidGregory084** [comment]
Aha! I think I have replicated this in #81 @mbesida! It seems it was fixed by the same change that fixed #74 so the reason that I could not replicate the issue is that is already resolved on `main`. I'm going to cut a release so that we can find out whether it resolve the issue for you.

**Comment by mbesida** [comment]
@DavidGregory084 that work's 👌.  Thanks a lot!
